src/dune_engine/action_intf.ml

```diff
@@ -11,6 +11,83 @@ module Inputs = struct
   type t = Stdin
 end
 
+module Memoize_or_distribute = struct
+  type t =
+    | Neither
+    | Memoize
+    | Distribute
+end
+
+module Simplified = struct
+  type destination =
+    | Dev_null
+    | File of string
+
+  type source = string
+
+  type t =
+    | Run of string * string list
+    | Chdir of string
+    | Setenv of string * string
+    | Redirect_out of t list * Outputs.t * destination
+    | Redirect_in of t list * Inputs.t * source
+    | Pipe of t list list * Outputs.t
+    | Sh of string
+end
+
+(** CR-soon cwong: It would be nice to not need to have [Action_ext_intf].
```

Comment:
This comment seems outdated because I can't see `Action_ext_intf` anywhere. Can you please update it and also make it a bit clearer what concretely is the problem?

Reply:
`Action_ext_intf` is in `dune_engine`.

Reply:
D'oh. Sorry, I don't know where I searched.
I still don't really understand why the current structure is problematic, but ok.

Reply:
I think that the `action_` set of modules is already convoluted and difficult-to-read as it is, but to some extent I don't think there's a good way around it.

Reply:
Err, and that adding more transforms and types that need to be looked at isn't helping*

Reply:
Oh, ok, got it! 
The problem is that the types `Action_ext_intf.{context,env}` are different from `Action_exec.{exec_context,exec_environment}` for no good reason. I have to say it wasn't clear to me from this CR.

Let me think about it for a bit.

Reply:
Two thoughts:

1) Maybe it's a good thing that those types are different. The dynamic dependencies stuff may not be a very tight abstraction in this context.

2) I think the definitions can be moved straightforwardly from Action_ext_intf to Action_intf.Ext with no cyclic dependendencies being introduced. I just tried it and it's fine. It makes for one fewer module names. Would you like me to push that change?

Reply:
That would be fine, sure!